src/screens/NewSubmission.tsx

```diff
@@ -43,25 +43,28 @@ export default function NewSubmissionScreen({ navigation }: DrawerScreenProps<Dr
     resolver: zodResolver(schema),
   });
 
-  const onSubmit: SubmitHandler<NewSubmissionFormData> = (data) => {
-    setIsLoading(true);
-    axiosInstance
-      .post('/submissions', data)
-      .then(response => {
-        setApiError(null);
-        navigation.navigate('HomeStack', {
-          screen: 'Home',
-          params: { newSubmissionAdded: true },
-        });
-        reset();
-      })
-      .catch(error => {
-        console.log(error.response);
-        const message = getErrorMessage(error);
-        setApiError(message);
-      }).finally(() => {
-        setIsLoading(false);
+
+  const createSubmission = (data: NewSubmissionFormData) => axiosInstance.post('/submissions', data);
+
+  const { mutate: createSubmissionMutation, isLoading } = useMutation({
```

Comment:
you can destructure the mutation error here 
```suggestion
  const { mutate: createSubmissionMutation, isLoading, error } = useMutation({
```

Reply:
this way you no longer need `const [apiError, setApiError] = useState<string | null>(null);`